mrmustard/lab_dev/states/base.py

```diff
@@ -351,6 +333,23 @@ def fock_distribution(self, cutoff: int) -> ComplexTensor:
         else:
             return math.reshape(math.abs(math.diag_part(fock_array)), (-1,))
 
+    def normalize(self) -> State:
+        r"""
+        Returns a rescaled version of the state such that its probability is 1.
+        """
+        probability = self.probability
+        if probability.shape != () and isinstance(self.ansatz, PolyExpAnsatz):
+            delta = len(self.ansatz.c.shape) - len(probability.shape)
+            probability = math.reshape(probability, probability.shape + (1,) * delta)
+        elif probability.shape != () and isinstance(self.ansatz, ArrayAnsatz):
+            probability = math.reshape(
+                probability, probability.shape + (1,) * self.ansatz.core_dims
+            )
+        if self.wires.ket and not self.wires.bra:
```

Comment:
```suggestion
        if not self.wires.ket or not self.wires.bra:
```
so that it works for adjoint states too